src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -1129,16 +1139,13 @@ public function validate(ExecutionContextInterface $context)
             }
         }
 
-        $coldStorageViolationMsg = 'You must provide both File Size and Sha256 Hash value for Cold Storage Information.';
-        if (null !== $this->datasetFileColdStorageArchiveSize && null === $this->datasetFileColdStorageArchiveSha256Hash) {
-                $context->buildViolation($coldStorageViolationMsg)
-                    ->atPath('datasetFileColdStorageArchiveSha256Hash')
-                    ->addViolation();
-        }
-
-        if (null === $this->datasetFileColdStorageArchiveSize && null !== $this->datasetFileColdStorageArchiveSha256Hash) {
+        $coldStorageViolationMsg = 'You must provide file size, sha256 hash, and original '
+        . 'filename for Cold Storage Information.';
+        if (false === (PelagosUtilities::nullOrNone(array($this->this->datasetFileColdStorageArchiveSize,
```

Comment:
This is a lot of code that could be done with a simple if statement. Also, a whole new class and utility for a function that will likely never get used again? 

`$test1 = array(null, 1, null);
$test2 = array(null, null, null);
$test3 = array(1, 1, 1);

var_dump(!array_filter($test1) or !in_array(null,$test1));
var_dump(!array_filter($test2) or !in_array(null,$test2));
var_dump(!array_filter($test3) or !in_array(null,$test3));`